h5p-bildetema/src/components/LanguageSelectorElement/LanguageSelectorElement.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { useL10ns } from "use-h5p";
+import { Language, UserData } from "../../../../common/types/types";
+import { defaultFavouriteLanguages } from "../Bildetema/Bildetema";
+import { languages, languagesOriginal } from "../../constants/languages";
+import { AllowedLanguage } from "../../types/AllowedLanguage";
+import styles from "./LanguageSelectorElement.module.scss";
+
+type LanguageSelectorElement = {
+  language: Language;
+  middleElement: boolean;
+  userData: UserData;
+  setUserData: (updatedUserData: UserData) => void;
+  favLanguages: Language[];
+  setFavLanguages: React.Dispatch<React.SetStateAction<Language[]>>;
+  handleChangeLanguage: (newLanguage: Language) => void;
+};
+
+export const LanguageSelectorElement: React.FC<LanguageSelectorElement> = ({
+  language,
+  middleElement,
+  userData,
+  setUserData,
+  favLanguages,
+  setFavLanguages,
+  handleChangeLanguage,
+}) => {
+  const languageKeys = languages.map(
+    lang => `lang_${lang}`,
+  ) as Array<`lang_${AllowedLanguage}`>;
+
+  const translations = useL10ns(...languageKeys, "selectLanguage");
+
+  const [isChecked, setIsChecked] = React.useState(
+    !!favLanguages.find(favLang => favLang.code === language.code),
+  );
+
+  const handleChange = (): void => {
+    const userDataSnapshot = userData;
+
+    if (
+      userDataSnapshot.favouriteLanguages.find(
+        favLang => favLang.code === language.code,
+      )
+    ) {
+      userDataSnapshot.favouriteLanguages =
+        userDataSnapshot.favouriteLanguages.filter(
+          favLang => favLang.code !== language.code,
+        );
+      setIsChecked(false);
+    } else {
+      userDataSnapshot.favouriteLanguages.push(language);
+    }
+
+    if (!userDataSnapshot.favouriteLanguages.length) {
+      userDataSnapshot.favouriteLanguages = defaultFavouriteLanguages;
+    }
+
+    setFavLanguages(userDataSnapshot.favouriteLanguages);
+    setUserData(userDataSnapshot);
+
+    if (
+      !userDataSnapshot.favouriteLanguages.find(
+        favLang => favLang.code === userDataSnapshot.currentLanguage.code,
+      )
+    ) {
+      handleChangeLanguage(userDataSnapshot.favouriteLanguages[0]);
+    }
+  };
+
```

Comment:
Do I understand this correctly?

```suggestion
    const currentLanguageWasUnfavorited = !userDataSnapshot.favouriteLanguages.find(
        favLang => favLang.code === userDataSnapshot.currentLanguage.code,
      );
    if (currentLanguageWasUnfavorited) {
```

Reply:
Yes! We need to deselect the current language if it was removed from the list of favorite languages, i.e. from the header, so a new language from favorites is selected instead.